tests/integration/100.command.dev.test.js

```diff
@@ -377,4 +378,95 @@ test(`catches invalid function names`, async (t) => {
     })
   })
 })
+
+test('should detect content changes in edge functions', async (t) => {
+  await withSiteBuilder('site-with-edge-functions', async (builder) => {
+    const publicDir = 'public'
+    await builder
+      .withNetlifyToml({
+        config: {
+          build: {
+            publish: publicDir,
+            edge_functions: 'netlify/edge-functions',
+          },
+          edge_functions: [
+            {
+              function: 'hello',
+              path: '/hello',
+            },
+          ],
+        },
+      })
+      .withEdgeFunction({
+        handler: () => new Response('Hello world'),
+        name: 'hello',
+      })
+
+    await builder.buildAsync()
+
+    await withDevServer({ cwd: builder.directory }, async ({ port }) => {
+      const helloWorldBody = () => got(`http://localhost:${port}/hello`).then((response) => response.body)
+
+      await builder
+        .withEdgeFunction({
+          handler: () => new Response('Hello builder'),
+          name: 'hello',
+        })
+        .buildAsync()
+
+      const DETECT_FILE_CHANGE_DELAY = 500
+      await pause(DETECT_FILE_CHANGE_DELAY)
+
+      const helloWorldMessage = await helloWorldBody()
+
+      t.is(helloWorldMessage, 'Hello builder')
+    })
+  })
+})
+
+test('should detect deleted edge functions', async (t) => {
+  await withSiteBuilder('site-with-edge-functions', async (builder) => {
+    const publicDir = 'public'
+    builder
+      .withNetlifyToml({
+        config: {
+          build: {
+            publish: publicDir,
+            edge_functions: 'netlify/edge-functions',
+          },
+          edge_functions: [
+            {
+              function: 'auth',
+              path: '/auth',
+            },
+          ],
+        },
+      })
+      .withEdgeFunction({
+        handler: () => new Response('Auth response'),
+        name: 'auth',
+      })
+
+    await builder.buildAsync()
+
+    await withDevServer({ cwd: builder.directory }, async ({ port }) => {
+      const authBody = () =>
```

Comment:
Similarly to the other test, consider making the HTTP call more explicit as opposed to using this intermediary function.